examples/with-typescript-types/README.md

```diff
@@ -0,0 +1,58 @@
+# TypeScript types Next.js example
+
+This is a really simple project that shows the usage of Next.js with TypeScript types;
+
+## Deploy your own
+
+Deploy the example using [Vercel](https://vercel.com):
+
+[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/import/project?template=https://github.com/vercel/next.js/tree/canary/examples/with-typescript-types)
+
+## How to use it?
+
+### Using `create-next-app`
+
+Execute [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app) with [npm](https://docs.npmjs.com/cli/init) or [Yarn](https://yarnpkg.com/lang/en/docs/cli/create/) to bootstrap the example:
+
+```bash
+npx create-next-app --example with-typescript-types with-typescript-types
+# or
+yarn create next-app --example with-typescript-types with-typescript-types
+```
+
+### Download manually
+
+Download the example:
+
+```bash
+curl https://codeload.github.com/vercel/next.js/tar.gz/canary | tar -xz --strip=2 next.js-canary/examples/with-typescript-types
+cd with-typescript-types
+```
+
+Install it and run:
+
+```bash
+npm install
+npm run dev
+# or
+yarn
+yarn dev
+```
+
+Deploy it to the cloud with [Vercel](https://vercel.com/import?filter=next.js&utm_source=github&utm_medium=readme&utm_campaign=next-example) ([Documentation](https://nextjs.org/docs/deployment)).
+
+## Notes
+
+This example shows how to integrate the TypeScript type system into Next.js. Since TypeScript is supported out of the box with Next.js, all we have to do is to install TypeScript.
+
+This example shows how to properly export and import typescript types without getting the
+
+```js
+Attempted import error: 'TypeA' is not exported from './package-1'.
+```
+
+error as addressed in [7882](https://github.com/vercel/next.js/issues/7882)
+
+## Useful links
+
+[Add import type and export type support to TypeScript](https://github.com/babel/babel/pull/11171)
```

Comment:
Let's replace the top description with this

Reply:
done

Reply:
@lfades done.